Engine/Colliders/Collider.cs

```diff
@@ -0,0 +1,49 @@
+﻿using AntEngine.Maths;
+
+namespace AntEngine.Colliders
+{
+    /// <summary>
+    /// Abstract collider allows for the calculation of collision between primitive geometrical shapes.
+    /// </summary>
+    public abstract class Collider
+    {
+        /// <summary>
+        /// Boundaries of the collider.
+        /// </summary>
+        private Transform Transform;
+
+        protected Collider() { }
+
+        /// <summary>
+        /// Check for a collision between this and a generic "collider" object.
+        /// Calls the checkCollision of the other object.
+        /// </summary>
+        /// <param name="collider"></param>
+        /// <returns></returns>
+        protected bool checkCollision(Collider collider)
+        {
+            return collider.checkCollision(this);
+        }
```

Comment:
This leads to StackOverflow because "this" refers to the Collider class and not the concrete ones (Circle, Square, etc.)

Reply:
Wouldn't we be better off just removing the non abstract "abstract" checkCollision and making the class an interface then? Or should Collider still be an abstract class?

Reply:
Yep seems better for the moment, we'll see later if checkCollider(Collider) is necessart